lib/csv_file_from_public_host.rb

```diff
@@ -16,7 +16,7 @@ def self.csv_response(path, env: ENV)
       connection.basic_auth(basic_auth_user, basic_auth_password)
     end
 
-    connection.get(path) do |req|
+    connection.get(Addressable::URI.escape(path)) do |req|
```

Comment:
Are we using Addressable here for a somewhat magic URI escape rather than say CGI.escape ? I'm guessing we're wanting backslashes to not be separated and not expecting other characters?

Just wondered, if my assumption is correct, if a) we need to document that and b) if we're risking any surprises - e.g. potential for weird user input to do anything fishy. 

I also wondered if already have the information escaped at all on the way in from Rails Request, but can't remember how much unescaping goes on - I assume a bit.

Reply:
@kevindew Good point. A more realistic test example would be a path containing slashes – because that's what happens in the real world.

I've used Addressable here because there's already a precedent for it being used to escape URLs:
https://github.com/alphagov/whitehall/blob/7a94656ccaaf030d7f2db590a77abda627f712f3/lib/whitehall/asset_manager_storage.rb#L36-L38

I did initially try with `URI.escape` but got warnings that it's an obsolete method.

Do you think I should be using `CGI.escape` instead?

Reply:
In fact, I've just found some prior art for this which I'll reference in the commit message for posterity.

Reply:
Yup good shout on not using `URI.escape`. 

`CGI.escape` wouldn't be a drop replacement as it will escape slashes. E.g: 

```
[1] pry(main)> CGI.escape("test/item-with-special-chars-%://.pdf")
=> "test%2Fitem-with-special-chars-%25%3A%2F%2F.pdf"
```

I always find myself a bit unsure if having Addressable do it smart introduces a fresh risk as it, presumably, has to guess a little as to what a special character means - this is me being a bit Captain Edgecase - so wondered if it was absolutely necessary. 

For what it's worth, even if I'm right and there's a potential edge case it'd probably not surface in actual use case.

Reply:
Oh yes, definitely. I've not introduced Addressable to the codebase – it looks like it's [already in use in Whitehall][whitehall], as well as [Asset Manager] and [GDS API adapters]. So I guess if there's an edge case, at least everything will fail consistently! 😬 

[whitehall]: https://github.com/alphagov/whitehall/blob/7a94656ccaaf030d7f2db590a77abda627f712f3/lib/whitehall/asset_manager_storage.rb#L36-L38
[Asset Manager]: https://github.com/alphagov/asset-manager/pull/396
[GDS API adapters]: https://github.com/alphagov/gds-api-adapters/pull/794/

Reply:
@kevindew would you mind taking a look again now, given this conversation and the extra detail I've added to the commit message? 🙏🏻